### DISCUSSION TITLE
How do I check whether the server is just started

### DISCUSSION THREAD

**Discussion by thebluetropics** [actix/actix-web#3264] [Q&A] [answered]
```rs
  let server = match HttpServer::new(move || {
    actix_web::App::new()
      .app_data(app.clone())
  }).bind(("127.0.0.1", 80)) {
    Ok(server) => server,
    Err(e) => {
      eprintln!("{}", e);
      bail!(Error::UnknownError);
    }
  };

  let _ = server.run().await;

  Ok(())
  ```
  
  In Node.js + Express, I can do this:
 ```
 const app = express()
 
 app.listen(80, function() {
   console.log("server is listening...")
 })
 ```
 
 How can I achieve the same result as this?
 Should I put it in the `Ok(server) => {}`?
 
 Or is there any other appropriate way to do this?

**Comment by robjtede** [comment, marked as answer]
```rust
// server is not up yet

let task_handle = tokio::spawn(server.run());

// server is now started and listening

println!("server is listening...")

let _ = task_handle.await;

// server has now shutdown
```

**Reply by thebluetropics** [reply]
Hi, thanks for the quick answer!

**Reply by Alirexaa** [reply]
> ```rust
> // server is not up yet
> 
> let task_handle = tokio::spawn(server.run());
> 
> // server is now started and listening
> 
> println!("server is listening...")
> 
> let _ = task_handle.await;
> 
> // server has now shutdown
> ```

This is not completely correct if the server failed to start the `server is listening...` is printed anyway, it's wrong.

for a quick run try this:

```rust
let server = match HttpServer::new(move || {
    actix_web::App::new()
      .app_data(app.clone())
  }).bind(("wronghostname", 80)) {
    Ok(server) => server,
    Err(e) => {
      eprintln!("{}", e);
      bail!(Error::UnknownError);
    }
  };
```

@robjtede, 
I want to print `server is ready to accept connections` after the server successfully starts and print an error if it failed to start.
how can I handle this situation?
